File: server/apps/shop/api/serializers/promotional_video.py
from rest_framework import serializers

from ...models import PromotionalVideo
from ..mixins import TranslationMixin


class PromotionalVideoTranslationMixin(TranslationMixin):
    """Миксин для переводов полей promotional video"""

    def get_title(self, obj):
        return self.get_translated_field(obj, "title")

    def get_subtitle(self, obj):
        return self.get_translated_field(obj, "subtitle")


class PromotionalVideoSerializer(
    PromotionalVideoTranslationMixin, serializers.ModelSerializer
):
    title = serializers.SerializerMethodField()
    subtitle = serializers.SerializerMethodField()
    video_id = serializers.CharField(source="youtube_video_id", read_only=True)

    class Meta:
        model = PromotionalVideo
        fields = (
            "id",
            "title",
            "subtitle",
            "youtube_url",
            "video_id",
            "background_image",
            "is_active",
            "order",
        )

File: intro-app/src/api/modules/promotional_video.js
import { publicApiClient } from "../client";
import { API_ENDPOINTS } from "../../constants/constants";

export const PromotionalVideoApi = {
    async get() {
        try {
            const { data } = await publicApiClient.get(
                API_ENDPOINTS.modules.promotional_video
            );

            if (data?.data) {
                const video = data.data;
                
                return {
                    id: video.id,
                    title: video.title,
                    subtitle: video.subtitle,
                    videoUrl: video.youtube_url,
                    backgroundImage: video.background_image,
                    isActive: video.is_active,
                    order: video.order,
                };
            }

            return null;
        } catch (error) {
            throw error;
        }
    }
};
